src/target/riscv/riscv.c

```diff
@@ -2323,6 +2323,15 @@ int riscv_init_registers(struct target *target)
 				case CSR_SATP:
 					r->exist = riscv_supports_extension(target, 'S');
 					break;
+				case CSR_MEDELEG:
+				case CSR_MIDELEG:
+					/* "In systems with only M-mode, or with both M-mode and
+					 * U-mode but without U-mode trap support, the medeleg and
+					 * mideleg registers should not exist." */
+					r->exist = (riscv_supports_extension(target, 'S') ||
```

Comment:
this should be `r->exist = (supports(S) || (!supports(S) && supports (N)))`. If you have `N`, you SHOULD have this register, which is not what the code currently does.

Reply:
Right. I tried to derive this again. Can you check my logic, starting with what it says in the spec (I think):
                    r->exist = !((M && !U && !S) || (M && U && !N))
                             = !(M && !U && !S) && !(M && U && !N)
                             = (!M || U || S) && (!M || !U || N)
                             = (U || S) && (!U || N)

Reply:
I think the phrase "or with both M-Mode and U-Mode  but without U-mode trap support" has an implied "only"... as in, "only M-Mode and U-Mode, but without U-mode trap support".

Also there are other implied things like you can't have S without U, and you can't have N without U.